rust/processor/src/parquet_manager.rs

```diff
@@ -0,0 +1,554 @@
+                let cloned_file = file.try_clone()?;
+                writer = SerializedFileWriter::new(cloned_file, schema.clone(), Arc::new(props.clone()))?;
+
+                info!("self file path after upload and reset: {:?}", file_path);  
+
+                info!("[Parquet Handler]Sending result to gap detector...");
+                gap_detector_sender
+                    .send(processing_result)
+                    .await
+                    .expect("[Parser] Failed to send versions to gap detector");
+            }
+        }
+        
+        let _processing_duration_in_secs: f64 = processing_start.elapsed().as_secs_f64();
+
+        Ok(ParquetProcessingResult {
+            start_version : 0, //move_resources.first().unwrap().transaction_version.clone() as u64,
+            end_version : 0, // move_resources.last().unwrap().transaction_version.clone() as u64,
+            // processing_duration_in_secs,
+            last_transaction_timestamp,
+            parquet_insertion_duration_in_secs: 0.0,
+            txn_version_to_struct_count: AHashMap::new(),
+        })
+    }
+
+    async fn handle_move_resource(
+        &mut self,
+        gcs_client: &GCSClient,
+        gap_detector_sender: &kanal::AsyncSender<ParquetProcessingResult>, 
+        changes: ParquetData) -> Result<ParquetProcessingResult>
+    {
+        let processing_start: std::time::Instant = std::time::Instant::now();
+
+        let props = WriterProperties::builder()
+            .set_compression(parquet::basic::Compression::LZ4) 
+            .build();
+        let props_arc = Arc::new(props.clone());
+        let last_transaction_timestamp = changes.last_transaction_timestamp;
+        let parquet_structs = changes.data.clone();
+        let txn_version_to_struct_count = changes.transaction_version_to_struct_count.clone();
+        self.transaction_version_to_struct_count.extend(txn_version_to_struct_count.clone());
+
+        let schema = get_schema(parquet_structs.first().unwrap().clone());
+        let table_name = get_file_path_str_from_struct(&parquet_structs.clone().first().unwrap().clone());
+
+        let file_path = table_name.clone() + ".parquet";
+        info!("opening the file at path: {:?}", file_path.clone());
+        let mut file: File = File::options().create(true).write(true).open(&file_path)?;
+        if self.writer.is_none() {
+            self.writer = Some(SerializedFileWriter::new(file.try_clone()?, schema.clone(), props_arc)?);
+        }
+
+        // let file_cloned = file.try_clone()?;
+        let mut writer: SerializedFileWriter<File> = self.writer.take().unwrap(); // Take out the writer temporarily
+        info!("Current buffer size: {:?}", self.buffer_size_bytes);
+
+        // loop the vector of struct and write to the parquet file 
+        for parquet_struct in parquet_structs.clone() {
+            self.buffer.push(parquet_struct.clone());
+            self.update_buffer_size(&parquet_struct);
```

Comment:
this seems to be duplicated- maybe some kind of generic method instead?

Also when you open the file, do you want to truncate if it already exists?

Reply:
yes, we should truncate if it already exists to avoid excessive duplicates